src/DataProtection/DataProtection/src/AuthenticatedEncryption/IAuthenticatedEncryptor.cs

```diff
@@ -32,5 +32,46 @@ public interface IAuthenticatedEncryptor
         /// <returns>The ciphertext blob, including authentication tag.</returns>
         /// <remarks>All cryptography-related exceptions should be homogenized to CryptographicException.</remarks>
         byte[] Encrypt(ArraySegment<byte> plaintext, ArraySegment<byte> additionalAuthenticatedData);
+
+#if NETCOREAPP
+        /// <summary>
+        /// Returns true if this implementation supports the span enabled versions of TryEncrypt/Decrypt
+        /// </summary>
+        /// <returns>true if this implementation supports the span enabled versions of TryEncrypt/Decrypt.</returns>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0016:Add public types and members to the declared API", Justification = "<Pending>")]
+        bool IsSpanEnabled() => false;
```

Comment:
This feels weird.

Reply:
The existing pattern was to have an internal derived interface IOptimizedAuthenticatedEncryptor, which was the previous iterations equivalent of an internal ISpanAuthenticatedEncryptor.  Then KeyRingDataProtector would try casting as opposed to calling IsSpanEnabled.  I'm ambivalent between the two do you have a strong preference?  The downside of the internal interface is there's no way for anyone else to implement Span enabled encryptors which seems pretty bad